agent/src/main/java/com/microsoft/applicationinsights/agent/internal/agent/AgentImplementation.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * ApplicationInsights-Java
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.agent.internal.agent;
+
+import java.io.File;
+import java.io.UnsupportedEncodingException;
+import java.lang.instrument.Instrumentation;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.net.URLDecoder;
+import java.util.jar.JarFile;
+
+import com.microsoft.applicationinsights.agent.internal.logger.InternalAgentLogger;
+
+/**
+ * Created by gupele on 5/6/2015.
+ */
+public final class AgentImplementation {
+
+    private final static String AGENT_JAR_PREFIX = "applicationinsights-agent";
+
+    private static String agentJarLocation;
+
+    public static void premain(String args, Instrumentation inst) {
+
+        try {
+            agentJarLocation = getAgentJarLocation();
+            appendJarsToBootstrapClassLoader(inst);
+            loadJarsToBootstrapClassLoader(inst);
+        } catch (Throwable throwable) {
+            InternalAgentLogger.INSTANCE.error("Agent is NOT activated: failed to load to bootstrap class loader: " + throwable.getMessage());
+            System.exit(-1);
```

Comment:
I think so, otherwise the user might get the impression that things are ok and will be surprised that the events are not in the portal, I think the 'fail-fast' approach is better